presto-main/src/test/java/com/facebook/presto/type/TestTimeBase.java

```diff
@@ -214,4 +214,82 @@ public void testIndeterminate()
         assertOperator(INDETERMINATE, "cast(null as TIME)", BOOLEAN, true);
         assertOperator(INDETERMINATE, "TIME '00:00:00'", BOOLEAN, false);
     }
+
+    @Test
+    public void testDistinctFrom()
+    {
+        assertFunction("TIME '12:34:56' IS DISTINCT FROM TIME '12:34:56'", BOOLEAN, false);
+        assertFunction("TIME '12:34:56.1' IS DISTINCT FROM TIME '12:34:56.1'", BOOLEAN, false);
+    }
+
+    @Test
+    public void testLessThanOrEquals()
+    {
+        // equality
+        assertFunction("TIME '12:34:56' <= TIME '12:34:56'", BOOLEAN, true);
+        assertFunction("TIME '12:34:56.1' <= TIME '12:34:56.1'", BOOLEAN, true);
+        assertFunction("TIME '12:34:56.12' <= TIME '12:34:56.12'", BOOLEAN, true);
+        assertFunction("TIME '12:34:56.123' <= TIME '12:34:56.123'", BOOLEAN, true);
+
+        // less than
+        assertFunction("TIME '12:34:56' <= TIME '12:34:57'", BOOLEAN, true);
+        assertFunction("TIME '12:34:56.1' <= TIME '12:34:56.2'", BOOLEAN, true);
+        assertFunction("TIME '12:34:56.12' <= TIME '12:34:56.13'", BOOLEAN, true);
+        assertFunction("TIME '12:34:56.123' <= TIME '12:34:56.124'", BOOLEAN, true);
+
+        // false cases
+        assertFunction("TIME '12:34:56' <= TIME '12:34:55'", BOOLEAN, false);
+        assertFunction("TIME '12:34:56.1' <= TIME '12:34:56.0'", BOOLEAN, false);
+        assertFunction("TIME '12:34:56.12' <= TIME '12:34:56.11'", BOOLEAN, false);
+        assertFunction("TIME '12:34:56.123' <= TIME '12:34:56.122'", BOOLEAN, false);
+    }
+
+    @Test
+    public void testGreaterThanOrEquals()
+    {
+        // equality
+        assertFunction("TIME '12:34:56' >= TIME '12:34:56'", BOOLEAN, true);
```

Comment:
Regarding Greater than, less than etc Most of these test cases are already tested in the same file above

Reply:
its tested for timestamp and in that we dnt have the same api, please confirm